src/tiff.ml

```diff
@@ -274,6 +281,18 @@ module Ifd = struct
     | ADOBE_DEFLATE -> 8
     | Other i -> i
 
+  type predictor = No_predictor | HorizontalDifferencing | Unknown of int
```

Comment:
Interesting interface here; how performance sensitive is this extraction? It could be stored unboxed, but not with this representation. (you can only unbox a single-constructor directly).

If this predictor is per-cell, you could try a private int type

```
module Predictor = struct
  type t = int
  let no_predictor x = (x = 1)
  let horizontal_differencing x = (x = 2)
end : sig
  type t = private int
  ...
end
```

but it depends how the predictors are used by the client

Reply:
Mostly it's an internal thing, and will be used to select a decompression scheme, so not per cell, rather once per bulk data read.

Reply:
This is only happening once per TIFF IFD (if at all, as we only compute the nice type if the user requests it) so I don't think we need to be optimising it just yet :)